hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/security/RouterSecurityManager.java

```diff
@@ -62,12 +67,14 @@ public RouterSecurityManager(Configuration conf) throws IOException {
         throw new IOException("Failed to create SecretManager");
       }
     }
+    auditLogger = new RouterSecurityAuditLogger(conf);
   }
 
   @VisibleForTesting
   public RouterSecurityManager(AbstractDelegationTokenSecretManager
       <DelegationTokenIdentifier> dtSecretManager) {
     this.dtSecretManager = dtSecretManager;
+    auditLogger = new RouterSecurityAuditLogger(new Configuration());
```

Comment:
doing new Configuration() is usually a bad idea, This method is used only in test, explore deprecating this and add a new method which takes in the conf and pass it from the test.

In the new method if ``conf`` is ``null`` don't ``initialise`` the ``auditLogger`` instead just put a ``warn`` log that ``RouterAuditLogger`` is ``disabled and something like that